python/kdumpfile.c

```diff
@@ -101,35 +103,59 @@ kdumpfile_dealloc(PyObject *_self)
 	self->ob_type->tp_free((PyObject*)self);
 }
 
+static PyObject *exception_map(kdump_status status)
+{
+	switch (status) {
+	case kdump_ok:		return NULL;
+	case kdump_syserr:	return PyExc_SystemError;
```

Comment:
`SystemError` is raised when the interpreter finds an internal error. You probably wanted to use `OSError`, because, indeed, errno is always set when libkdumpfile returns this status code. OTOH, I'm not sure if (and how) Python can combine the `errno` value, default `strerror` and additional info about the error condition (available through `kdump_err_str`).

To illustrate my point, I've got a corrupt dump file, where the call to kdump_set_fd() fails with:
- status `kdump_syserr`
- `errno` is set to `ENOMEM`
- `kdump_err_str()` returns: "Cannot allocate VMCOREINFO buffer (18446612132379117088 bytes)"